### PR TITLE
`catalog_product_entity_media_gallery` not cleared when related products deleted.

### LINKED ISSUE #17727: `catalog_product_entity_media_gallery` not cleared when related products deleted.
### Preconditions
1. Magento 2.2.5CE with sample data. 

### Steps to reproduce
1. Go to Products page and delete all products.
2. Look at `catalog_product_entity_media_gallery` table.

### Expected result

1. Table `catalog_product_entity_media_gallery` is empty.

### Actual result

1. It's still has 3.422 rows.  There is no row was deleted.

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#17727

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/Catalog/Model/ResourceModel/Product/GalleryTest.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        $product = $this->productRepository->get('simple');
+
+        $attributeId = $this->readHandler->getAttribute()->getAttributeId();
+        $mediaGalleryData = $this->galleryResource->loadProductGalleryByAttributeId($product, $attributeId);
+        $values = array_column($mediaGalleryData, 'value_id');
+
+        $this->productRepository->delete($product);
+
+        $this->assertEmpty($this->getMediaGalleryDataByValues($values));
```
Comment (ihor-sviziev) [inline dev/tests/integration/testsuite/Magento/Catalog/Model/ResourceModel/Product/GalleryTest.php:75]: Can we make sure that before deleting, there were some gallery data?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: dev/tests/integration/testsuite/Magento/Catalog/Model/ResourceModel/Product/GalleryTest.php (hunk 1 above)
Comment (engcom-Charlie) [reply dev/tests/integration/testsuite/Magento/Catalog/Model/ResourceModel/Product/GalleryTest.php:75]: Sure.

**Comment by engcom-Charlie** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/Catalog/Model/ResourceModel/Product/GalleryTest.php (hunk 1 above)
Comment (engcom-Charlie) [reply dev/tests/integration/testsuite/Magento/Catalog/Model/ResourceModel/Product/GalleryTest.php:75]: @ihor-sviziev done.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8814](https://jira.corp.magento.com/browse/ENGCOM-8814) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

1. Go to Products page and delete all products.
2. Look at `catalog_product_entity_media_gallery` table.

**Before:** :heavy_multiplication_x: It still has rows with images.
Screenshot from 2021-02-25 12-14-26

**After:** :heavy_check_mark: Table `catalog_product_entity_media_gallery` is empty.
Screenshot from 2021-02-25 12-27-54

Was checked with Old and New Media Gallery and everything works as expected.

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Pull request merged**